### PR TITLE
MON-106121-gorgone-use-vault-autonomous-library

### PR DESCRIPTION
**Pull request by Evan-Adam**
## Description

Add centreonvault dependancy and use it to retrieve all secret in the gorgone conf with a vault API.

**Fixes** # MON-106121

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [X] master

<h2> How this pull request can be tested ? </h2>

check gorgone can still be launched and correctly write it's log to the log file (start the service, check the log for any error, there should not be any)
configure vault and change one of gorgone property for a secret. The secret should be correctly retrieved from the vault.

## Checklist

- [X] I have followed the **coding style guidelines** provided by Centreon
- [X] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [X] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have made corresponding changes to the **documentation**.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/actions/package/action.yml [hunk 1]
```diff
@@ -116,6 +116,6 @@ runs:
       name: Upload package artifacts
       uses: actions/upload-artifact@0b2256b8c012f0828dc542b3febcab082c67f72b # v4.3.4
       with:
-        name: ${{ inputs.arch != '' && format('packages-{0}-{1}', inputs.distrib, inputs.arch) || format('packages-{0}', inputs.distrib) }}
+        name: ${{ inputs.stability }} ${{ inputs.arch != '' && format('packages-{0}-{1}', inputs.distrib, inputs.arch) || format('packages-{0}', inputs.distrib) }}
```
Comment (mushroomempires) [inline .github/actions/package/action.yml:119]:
```suggestion
        name: ${{ inputs.arch != '' && format('packages-{0}-{1}', inputs.distrib, inputs.arch) || format('packages-{0}', inputs.distrib) }}-${{ inputs.stability }}
```

File: .github/workflows/gorgone.yml [hunk 2]
```diff
@@ -16,7 +16,6 @@ on:
       - ".github/workflows/gorgone.yml"
       - "gorgone/**"
       - "perl-libs/**"
-      - "!gorgone/tests/**"
```
Comment (mushroomempires) [inline .github/workflows/gorgone.yml:19]: to revert

File: .github/workflows/gorgone.yml [hunk 3]
```diff
@@ -29,7 +28,6 @@ on:
...
```
Comment (mushroomempires) [inline .github/workflows/gorgone.yml:32]: to revert

**Review by mushroomempires — APPROVED** [review]

File: gorgone/tests/unit/class/core.t [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    my $set = shift;
+    # let's make a simple object and try to industryalize the yaml read configuration.
+    my $gorgone        = gorgone::class::core->new();
+    $gorgone->{logger} = $set->{logger};
+    $gorgone->{vault} = centreon::common::centreonvault->new();
+
+    my $tests_cases = [
+        {
+            file => './config_examples/simple_no_recursion/norecursion.yaml',
+            got  => { configuration => { gorgone => {
```
Comment (omercier) [inline gorgone/tests/unit/class/core.t:62]:
```suggestion
            expected  => { configuration => { gorgone => {
```

File: gorgone/tests/unit/class/core.t [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                vault  => {
+                    badFormat     => 'secret::hashicorp::thereIsOnlyOneColon',
+                    correctFormat => 'VaultSentASecret'},
+
+            } } },
+            msg  => 'simple configuration without recursion'
+        },
+        {
+            file => './config_examples/include_other_files/main.yaml',
+            got  => { configuration => { gorgone => {
```
Comment (omercier) [inline gorgone/tests/unit/class/core.t:76]:
```suggestion
            expected  => { configuration => { gorgone => {
```

File: gorgone/tests/unit/class/core.t [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+                gorgonecore => { global_variable => "value" }
+            } } },
+            msg  => 'simple configuration with !include.'
+        },
+        { # this is a real world exemple with all parameter I could think of. The default configuration don't have all of them.
+            # this is more an integration test than a unit test, but allow to test the whole configuration.
+            file => './config_examples/centreon-gorgone/config.yaml',
+            got  => require("./config_examples/centreon-gorgone/expectedConfiguration.pl"),
```
Comment (omercier) [inline gorgone/tests/unit/class/core.t:84]:
```suggestion
            expected  => require("./config_examples/centreon-gorgone/expectedConfiguration.pl"),
```

**Review by bouda1 — APPROVED** [review]

File: gorgone/docs/configuration.md [hunk 7]
```diff
@@ -100,6 +100,17 @@ configuration:
       proxy_name: proxy
 ```
 
+## *centreon vault*
+
+Centreon vault is a tool to secure passwords present in the Centreon configuration.\
```
Comment (cg-tw) [inline gorgone/docs/configuration.md:105]:
```suggestion
Centreon Vault is a tool that secures the passwords present in the Centreon configuration.\

```

File: gorgone/docs/configuration.md [hunk 8]
```diff
@@ -100,6 +100,17 @@ configuration:
       proxy_name: proxy
 ```
 
+## *centreon vault*
+
+Centreon vault is a tool to secure passwords present in the Centreon configuration.\
+It allows to store the password in a vault and retrieve it when needed by each component.\
```
Comment (cg-tw) [inline gorgone/docs/configuration.md:106]:
```suggestion
It stores passwords in a vault and retrieves them when needed by each component.\

```

File: gorgone/docs/configuration.md [hunk 9]
```diff
@@ -100,6 +100,17 @@ configuration:
       proxy_name: proxy
 ```
 
+## *centreon vault*
+
+Centreon vault is a tool to secure passwords present in the Centreon configuration.\
+It allows to store the password in a vault and retrieve it when needed by each component.\
+Gorgone allow to use a vault to store any string in the configuration. It cannot store an array or an hash.
```
Comment (cg-tw) [inline gorgone/docs/configuration.md:107]:
```suggestion
Gorgone allows you to use a vault to store any string in the configuration. It cannot store an array or a hash.

```

File: gorgone/docs/configuration.md [hunk 10]
```diff
@@ -100,6 +100,17 @@ configuration:
       proxy_name: proxy
 ```
 
+## *centreon vault*
+
+Centreon vault is a tool to secure passwords present in the Centreon configuration.\
+It allows to store the password in a vault and retrieve it when needed by each component.\
+Gorgone allow to use a vault to store any string in the configuration. It cannot store an array or an hash.
+
+To use vault, report to the official documentation to set up vault and the configuration file in `/var/lib/centreon/vault/vault.json`\
```
Comment (cg-tw) [inline gorgone/docs/configuration.md:109]:
```suggestion
To use Vault, read the official documentation to set up Vault and the configuration file in `/var/lib/centreon/vault/vault.json`\

```

File: gorgone/docs/configuration.md [hunk 11]
```diff
@@ -100,6 +100,17 @@ configuration:
...
 ```
 
+## *centreon vault*
+
+Centreon vault is a tool to secure passwords present in the Centreon configuration.\
+It allows to store the password in a vault and retrieve it when needed by each component.\
+Gorgone allow to use a vault to store any string in the configuration. It cannot store an array or an hash.
+
+To use vault, report to the official documentation to set up vault and the configuration file in `/var/lib/centreon/vault/vault.json`\
+Then replace any password present in the gorgone configuration with a vault string. See the official format here :
```
Comment (cg-tw) [inline gorgone/docs/configuration.md:110]:
```suggestion
Then replace any password present in the Gorgone configuration with a Vault string. See the official format here : 

```

**Review by cg-tw — APPROVED** [review]

**Pull request merged**